src/services/course.js

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable import/prefer-default-export */
+import { Op } from 'sequelize';
+import parse from 'csv-parse/lib/sync';
+import db, { sequelize } from '../models';
+
+const {
+  Course,
+} = db;
+
+export async function csvImport(buffer) {
+  const created = [];
+  const replaced = [];
+  const updated = [];
+  const deleted = [];
+  const errors = [];
+  const skipped = [];
+
+  const importedCourseIds = [];
+
+  const parsed = parse(buffer, { skipEmptyLines: true, columns: true });
+  let rowCount = 1;
+  let results;
+  try {
+    results = await Promise.all(parsed.map(async (course) => {
+      // Get the first property value form course object.
+      let rawCourseName = course['course name'];
+
+      if (!rawCourseName) {
+        // Skip blank course name.
+        skipped.push(`Row ${rowCount}: Blank course name`);
+        return false;
+      }
+
+      // Always trim leading and trailing spaces.
+      rawCourseName = rawCourseName.trim();
+
+      // Remove all spaces and special characters from the course name.
+      const cleanCourseName = rawCourseName.replace(/[^a-zA-Z0-9]/g, '').toLowerCase();
+
+      // Find all existing courses that match the clean course name regardless of case.
+      const existingCourses = await Course.findAll({
```

Comment:
This removed the need for the nameLookUp column
```suggestion
      const existingCourses = await Course.findAll({
  where: {
    [Op.and]: [
      Sequelize.where(
        Sequelize.fn('lower', Sequelize.fn('regexp_replace', Sequelize.col('name'), '[^a-zA-Z0-9]', '', 'g')),
        { [Op.like]: cleanCourseName }
      ),
      { deletedAt: null }
    ]
  }
});
```

Reply:
Updated to inline regex.